sonaranalyzer-dotnet/src/SonarAnalyzer.VisualBasic/Rules/MarkWindowsFormsMainWithStaThread.cs

```diff
@@ -0,0 +1,56 @@
+﻿/*
+ * SonarAnalyzer for .NET
+ * Copyright (C) 2015-2018 SonarSource SA
+ * mailto: contact AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System.Collections.Immutable;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.VisualBasic;
+using Microsoft.CodeAnalysis.VisualBasic.Syntax;
+using SonarAnalyzer.Common;
+using SonarAnalyzer.Helpers;
+
+namespace SonarAnalyzer.Rules.VisualBasic
+{
+    [DiagnosticAnalyzer(LanguageNames.VisualBasic)]
+    [Rule(DiagnosticId)]
+    public sealed class MarkWindowsFormsMainWithStaThread : MarkWindowsFormsMainWithStaThreadBase<MethodBlockSyntax>
+    {
+        private static readonly DiagnosticDescriptor rule =
+            DiagnosticDescriptorBuilder.GetDescriptor(DiagnosticId, MessageFormat, RspecStrings.ResourceManager);
+
+        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics { get; } = ImmutableArray.Create(rule);
+
+        protected override IMethodSymbol GetSymbol(SemanticModel model, MethodBlockSyntax method) =>
+            model.GetDeclaredSymbol(method);
+
+        protected override void Report(SyntaxNodeAnalysisContext c, MethodBlockSyntax method, string message) =>
+            c.ReportDiagnosticWhenActive(
+                Diagnostic.Create(rule,
+                    method.SubOrFunctionStatement.Identifier.GetLocation(),
```

Comment:
You could have an abstract method in the base class to get the identifier location. 

FYI instead of using `rule` when you will move this code to the base class you can use `SupportedDiagnostics[0]`.